genai-perf/genai_perf/config/generate/sweep_objective_generator.py

```diff
@@ -0,0 +1,187 @@
+                    ].get_objective_category(objective_name)
+                    usage = self._model_search_parameters[model_name].get_type(
+                        objective_name
+                    )
+
+                    model_objective_parameters[model_name][objective_name] = (
+                        ObjectiveParameter(usage, category, value)
+                    )
+
+            yield model_objective_parameters
+
+    ###########################################################################
+    # Parameter Combination Methods
+    ###########################################################################
+    def _create_list_of_model_search_parameter_combinations(
+        self, model_name: ModelName
+    ) -> ParameterCombinations:
+        search_parameters = self._model_search_parameters[model_name]
+
+        parameter_lists = {}
+        for parameter_name in search_parameters.get_parameter_names():
+            parameter_lists[parameter_name] = search_parameters.get_list(parameter_name)
+
+        parameter_names, parameter_combinations = zip(*parameter_lists.items())
+
+        model_search_parameter_combinations = [
+            dict(zip(parameter_names, parameter_combination))
+            for parameter_combination in product(*parameter_combinations)
+        ]
+
+        return model_search_parameter_combinations
+
+    def _create_list_of_all_search_parameter_combinations(
+        self,
+        model_search_parameter_combinations: ModelParameterCombinations,
+    ) -> AllParameterCombinations:
+        model_names, model_parameter_combinations = zip(
+            *model_search_parameter_combinations.items()
+        )
+
+        all_search_parameter_combinations = []
+        for model_parameter_combination in product(*model_parameter_combinations):
+            model_combination_dict = {}
+            for count, model_name in enumerate(model_names):
+                model_combination_dict[model_name] = model_parameter_combination[count]
+
+            all_search_parameter_combinations.append(model_combination_dict)
+
+        return all_search_parameter_combinations
+
+    ###########################################################################
+    # General Search Space Methods
+    ###########################################################################
+    def _calculate_num_of_configs_in_search_space(self) -> int:
+        num_of_configs_in_search_space = 1
+        for model_name in self._config.model_names:
+            num_of_configs_in_search_space *= self._model_search_parameters[
+                model_name
+            ].number_of_total_possible_configurations()
+
+        return num_of_configs_in_search_space
+
+    ###########################################################################
+    # Info/Debug Methods
+    ###########################################################################
+    def _print_debug_search_space_info(self) -> None:
+        logger.info("")
```

Comment:
Is this a debugging statement that was forgotten, or why do you need this?

Reply:
Are you referring to the `logger.info("")`? If so, those are intentional and create newlines around the debug info.

Reply:
The reason `info` (instead of `debug`) is used is that I wanted an actual newline vs. a line that was `DEBUG: `

Reply:
Could you use a newline in the quote below instead so that it's clearer what is being done? e.g.
```
num_of_configs_in_search_space = (
            self._calculate_num_of_configs_in_search_space()
        )
        logger.debug(
            f"\nNumber of configs in search space: {num_of_configs_in_search_space}"
        )
        self._print_debug_model_search_space_info()
```

And perhaps have another newline printed in the debug function? I guess I'm not sure why we want this. It seems misleading to me in terms of reading code and logs. Do you happen to have a screenshot of the logs (with/without these newlines) to help me understand the purpose here?

Reply:
I don't have a screenshot anymore, but with your suggestion it looks like:
`DEBUG:`
`DEBUG: <config_info>`
`DEBUG:`

versus with the code as is the newlines are actual newlines w/o printing the `DEBUG:` which to my eye just looks cleaner.

Reply:
Found an old PR from MA that highlights this: https://github.com/triton-inference-server/model_analyzer/pull/900
![image](https://github.com/user-attachments/assets/34a1ee1e-4f5f-4e80-9e3a-4bd0858470f8)

Reply:
Got it, thanks for explaining and providing a visual! I agree that this looks cleaner. Resolving this thread.